### PR TITLE
Deprecated database configuration removed

### PR DESCRIPTION
**Pull request by dirrao**
Deprecated database configuration removed

### CODE REVIEW

File: newsfragments/42126.significant.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+     * Removed deprecated configuration ``sql_alchemy_pool_size`` from ``core``. Please use ``sql_alchemy_pool_size`` from ``database`` instead.
+     * Removed deprecated configuration ``sql_alchemy_max_overflow`` from ``core``. Please use ``sql_alchemy_max_overflow`` from ``database`` instead.
+     * Removed deprecated configuration ``sql_alchemy_pool_recycle`` from ``core``. Please use ``sql_alchemy_pool_recycle`` from ``database`` instead.
+     * Removed deprecated configuration ``sql_alchemy_pool_pre_ping`` from ``core``. Please use ``sql_alchemy_pool_pre_ping`` from ``database`` instead.
+     * Removed deprecated configuration ``sql_alchemy_schema`` from ``core``. Please use ``sql_alchemy_schema`` from ``database`` instead.
+     * Removed deprecated configuration ``sql_alchemy_connect_args`` from ``core``. Please use ``sql_alchemy_connect_args`` from ``database`` instead.
+     * Removed deprecated configuration ``load_default_connections`` from ``core``. Please use ``load_default_connections`` from ``database`` instead.
+     * Removed deprecated configuration ``max_db_retries`` from ``core``. Please use ``max_db_retries`` from ``database`` instead.
```
Comment (uranusjr) [inline newsfragments/42126.significant.rst]: Does this indentation render correctly?

Comment (dirrao) [reply newsfragments/42126.significant.rst]: How to check it locally?

**Review by vincbeck — APPROVED** [review]

File: newsfragments/42126.significant.rst (hunk 1 above)
Comment (vincbeck) [reply newsfragments/42126.significant.rst]: Some reformatting will have to be done when releasing Airflow 3 anyway. I can see some differences in indentation in the difference significant files. We could actually use this PR to normalize them to, say, 2 spaces before the `*`?

Comment (dirrao) [reply newsfragments/42126.significant.rst]: I have updated this. I will address the other news fragment files in a different PR.

**Review by vincbeck — APPROVED** [review]

**Comment by amoghrajesh** [comment]
@dirrao was this supposed to be targetted for 2.10.3 also?
cc @zachliu

**Comment by dirrao** [comment]
> @dirrao was this supposed to be targetted for 2.10.3 also?
> cc @zachliu

@amoghrajesh This was intended for Airflow 3.

**Pull request merged**